### DISCUSSION TITLE
ADAL to MSAL migration - ID Token missing

### DISCUSSION THREAD

**Discussion by rkrsgithub** [AzureAD/microsoft-authentication-library-for-android#1694] [Q&A] [answered]
Hi Team,

I am trying to migrate from ADAL to MSAL. In MSAL `AuthenticationResult` ID token variable is missing. anyone here to help me the right way to get ID token in Authentication Result of MSAL.

Thanks in advance.
Raghu Krishnan R

**Comment by negoe** [comment, marked as answer]
While constructing PublicClientApplication, when you use MSAL, you instantiate a PublicClientApplication. This object models your app identity and is used to make requests to one or more authorities. With this object you will configure your client identity, redirect URI, default authority, whether to use the device browser vs. embedded web view, the log level, and more.

You can declaratively configure this object with JSON, which you either provide as a file or store as a resource within your APK.

Although this object is not a singleton, internally it uses shared Executors for both interactive and silent requests.

In ADAL, every organization that you request access tokens from requires a separate instance of the AuthenticationContext. In MSAL, this is no longer a requirement. You can specify the authority from which you want to request a token as part of your silent or interactive request.

_For more information you can follow these links:_ 
- [Migrate from AuthenticationContext to PublicClientApplications](https://learn.microsoft.com/en-us/azure/active-directory/develop/migrate-android-adal-msal)
- [How to use MSAL in your Code?](https://learn.microsoft.com/en-us/azure/active-directory/develop/tutorial-v2-android#use-msal)

**Comment by shahzaibj** [comment]
@rkrsgithub In MSAL, you can get ID Token as follows:

`authenticationResult.getAccount().getIdToken()`
